src/AutoBuild.jl

```diff
@@ -785,7 +785,7 @@ function autobuild(dir::AbstractString,
             verbose=verbose,
         )
         setup_deps(f, prefix, dependencies, platform, verbose) =
-            setup_dependencies(prefix, Pkg.Types.PackageSpec[getpkg(d) for d in filter_platforms(dependencies, platform) if f(d)], platform; verbose)
+            setup_dependencies(prefix, Pkg.Types.PackageSpec[getpkg(d) for d in filter_platforms(dependencies, platform) if f(d) && is_build_dependency(d)], platform; verbose)
```

Comment:
This was the main missing bit: we weren't filtering out non-build dependencies, because before https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/249 all dependencies were also build-dependencies.